code/unit_tests/items.dm

```diff
@@ -1,21 +1,34 @@
 /datum/unit_test/items_test
 	name = "ITEMS: Items Shall Pass Test Battery"
+	async = TRUE
 	var/list/obj_test_instances = list()
 	var/list/failures = list()
+	var/is_done = FALSE
 
 /datum/unit_test/items_test/start_test()
+	var/turf/spawn_loc = get_safe_turf()
+	var/list/cached_contents = spawn_loc.contents.Copy()
 	// Instantiate all spawnable items
 	for(var/path in subtypesof(/obj/item))
+		var/obj/item/I
 		try
-			var/obj/item/I = path
+			I = path
 			if(!TYPE_IS_SPAWNABLE(I))
 				continue
-			I = new path
+			I = new path(spawn_loc)
 			if(QDELETED(I))
 				continue
 			obj_test_instances[path] = I
+			// Move ourselves to nullspace to avoid interfering with stuff spawned after.
+			I.forceMove(null)
```

Comment:
I don't want errors moving or deleting those to cause the object itself to error; they should runtime outside of the try/catch IMO.

Reply:
That'd be moving it into the try/catch, though—so any runtimes while moving the stuff it might create would be falsely marked as a runtime during the creation of the object itself.

Reply:
the cleanup has given me no shortage of issues when i expanded it to cover more than just items. if it runtimes outside of the try/catch the whole test dies, which sucks, so i might just add a second try/catch